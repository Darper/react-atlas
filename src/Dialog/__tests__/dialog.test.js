import React from "react";
import { mount } from "enzyme";
import { Dialog } from "../index";

// function handleCancel() {
//   console(".-.");
// }

// TO-DO Add a snapshot test.

describe("Dialog component - Basic test", () => {
  it("Dialog component(info) - Basic test", function() {
    const dial = mount(
      <Dialog>
        <div>
          This is Dialog example<br />Any child components could be put here.
        </div>
      </Dialog>
    );
    expect(dial.props().active).toEqual(false);
  });

  it("Dialog component(confirm) - Basic test", function() {
    const handleToggle = jest.fn();
    const handleOk = jest.fn();
    const handleCancel = jest.fn();
    const dial = mount(
      <div>
        <button onClick={handleToggle}>Open</button>
        <Dialog
          active
          onOk={handleOk}
          onCancel={handleCancel}
          title="Confirm"
          overlay
          confirm
          className={"dialogClass"}
        >
          <div>
            This is Dialog example<br />Any child components could be put here.
          </div>
        </Dialog>
      </div>
    );
    dial.find("button").simulate("click");
    expect(
      dial
        .find(".dialogClass")
        .children()
        .first()
        .props().active
    ).toBe(true);
  });

  it("Dialog component(warning) - Basic test", function() {
    const handleToggle = jest.fn();
    const handleOk = jest.fn();
    const handleCancel = jest.fn();
    const dial = mount(
      <div>
        <button onClick={handleToggle}>Open</button>
        <Dialog
          active
          onOk={handleOk}
          onCancel={handleCancel}
          title="Warning"
          overlay
          warning
        >
          <div>
            This is Dialog example<br />Any child components could be put here.
          </div>
        </Dialog>
      </div>
    );
    expect(dial.find("Dialog").props().warning).toBe(true);
  });
});
