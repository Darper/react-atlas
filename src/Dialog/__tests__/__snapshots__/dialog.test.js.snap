// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dialog component Test render correctly 1`] = `null`;
